### PR TITLE
Use error modal for missing binaries

### LINKED ISSUE #2586: Empty log file downloaded if file does not exist anymore
## Describe the bug
If an Agent's log file is deleted, the download log button on the Map page's preview pane doesn't update (it should be gray if a log file doesn't exist), and when clicked, downloads an empty file of the same name.

## To Reproduce
1. Run the Island and an Agent. Go the Map page and click on the machine on which the Agent was run. See that the download Agent log button is yellow and click on it to verify that the file gets downloaded.
2. Delete that Agent's log file.
3. Go to the Map page. The button is still yellow and clicking on it download's an empty file with the same name.

## Expected behavior
The island should return a 404 error if a log doesn't exist. The user should get some kind of notification.

## Tasks
- [x] Ensure that the Island returns a 404 if the log doesn't exist (0d) - @shreyamalviya 
- [ ] Notify the user that the requested log file doesn't exist. (0.25d) - @shreyamalviya 
    - [ ] Preferably, always handle 404 errors in the UI the same way

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes #2586 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running the Island
* [x] If applicable, add screenshots or log transcripts of the feature working

image
image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunOnIslandButton.js [hunk 1]
```diff
@@ -96,14 +97,54 @@ class RunOnIslandButton extends AuthComponent {
...
             );
           }
         }
       });
   }
 
+  getErrorMessageFromErrorText(errorText) {
+    if (errorText.includes('Permission denied:') || errorText.includes('Text file busy')) {
+      return this.getMonkeyAlreadyRunningContent()
+    } else if (errorText.startsWith('Copy file failed')) {
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunOnIslandButton.js:112]: Why is the screenshot showing "You encountered an undefined error" if it obviously should be "Some Monkey binaries are not found where they should be..."? Can we improve the error message selection? This is valuable information for the user, AV can update any time and delete the binary, we need to be sure to handle this case properly

File: monkey/monkey_island/cc/ui/src/components/ui-components/ErrorModal.js [hunk 2]
```diff
@@ -32,16 +30,18 @@ class ErrorModal extends React.PureComponent {
         </Modal.Header>
         <Modal.Body>
           <div style={{'marginTop': '1em', 'marginBottom': '1em'}}>
-            <p className="alert alert-danger">
+            <div className="alert alert-danger">
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/ui-components/ErrorModal.js:33]: can we make the level a prop as well? not mandatory though

File: monkey/monkey_island/cc/ui/src/components/ui-components/ErrorModal.js [hunk 3]
```diff
@@ -11,8 +11,6 @@ class ErrorModal extends React.PureComponent {
 
     this.state = {
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/ui-components/ErrorModal.js:12]: Use hooks and typescript for this component

File: monkey/monkey_island/cc/ui/src/components/ui-components/ErrorModal.js [hunk 4]
```diff
@@ -32,16 +30,18 @@ class ErrorModal extends React.PureComponent {
...
             (<div>
-              <h4>Error Details</h4>
               <hr/>
-              <div>{this.state.errorDetails}</div>
+              <h4>Error Details</h4>
+              <p style={{'word-wrap': 'break-word'}}>
+                {this.props.errorDetails}
+              </p>
             </div>) : <></>
           }
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/ui-components/ErrorModal.js:46]: do this to reduce boilerplate:
```
{this.props.errorDetails !== undefined && <div> ... </div>}
```

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Most important: show the relevant error message and refactor to hooks+typescript

File: monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunOnIslandButton.js (hunk 1 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunOnIslandButton.js:112]: Good catch. Updated the message selection logic.
image

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**